bsmetadata/preprocessing_tools/html_parser/__init__.py

```diff
@@ -0,0 +1,36 @@
+from typing import List, Optional
+
+from bsmetadata.preprocessing_tools.html_parser.filters_and_cleaners import TextAndMetadataCleaner
+from bsmetadata.preprocessing_tools.html_parser.objects import TagToRemoveWithContent
+
+
+def get_clean_text_and_metadata(
+    html_str,
+    tags_to_remove_with_content: Optional[List[TagToRemoveWithContent]] = None,
+    tags_to_remove_alone: Optional[List[str]] = None,
+    attrs_to_keep: Optional[List[str]] = None,
+    consecutive_tags_to_fold: Optional[List[str]] = None,
+    convert_br_tag_to_breaking_line: Optional[bool] = False,
+    txt_max_chr_len_alone: float = -float("inf"),
+    txt_min_chr_len_alone: float = -float("inf"),
+    tags_exceptions_to_txt_max_min_chr_len_alone: List[str] = None,
+    txt_max_chr_len_with_content: float = -float("inf"),
+    txt_min_chr_len_with_content: float = -float("inf"),
```

Comment:
I don't fully understand the purpose of the variable, but shouldn't max be `inf` by default? (and min be `-inf`)

Also nit: I prefer`float("-inf")`  over `-float("inf")`

Reply:
All these `*max_chr_len*` and `*min_chr_len*` variables refer to conditions on the text to remove a tag or a tag and its content.

In this case, the tags whose content would be of length between `txt_min_chr_len_with_content` and `txt_max_chr_len_with_content`  would be removed from the html DOM .

So by setting both to `-float("inf")` the condition will never be verified and we will keep everything.

> Also nit: I preferfloat("-inf") over -float("inf")

I have no problem making this change but I would be curious to know why :slightly_smiling_face:

Reply:
That makes sense. Now I am curious why it's content of length within a range we want to remove, not outside of a range?

> > Also nit: I preferfloat("-inf") over -float("inf")

> I have no problem making this change but I would be curious to know why 🙂

Please Don't mind it. It's just I have never seen `-float("inf")` before, and my last brain cell warned me that it might be undefined. But now I think it's fine, it's just flipping one bit. No need to change it. 🙂

Reply:
It's true that what you propose makes sense, especially for this variable that would apply to all tags.

The reason why I arrived in this configuration is that I had started by introducing only the conditions of deletion of tag "case by case" in the list `tags_to_remove_with_content` and finally if we wanted to have the configuration that you propose it was "enough" to list twice the same tag but with conditions on the length of the content different (one for the lower bound and the other for the upper bound). I added that later on this condition for all the tags.

If we think that it is more useful to reverse the condition, maybe we can do it in a second PR, what do you think?

Reply:
> it was "enough" to list twice the same tag but with conditions on the length of the content different (one for the lower bound and the other for the upper bound)

That's a really clever design, I didn't realize we can list a tag more than once.

 > maybe we can do it in a second PR, what do you think? 🙂

I think we can merge this already, since the html test seems to work and this PR doesn't touch other part of the codebase. And I don't think we need to change that.

Thank you for working on this! 👍